### PR TITLE
Added a right-click option on the dock to don't import this directory

### PR DESCRIPTION
**Pull request by marcelofg55**
This PR adds a right-click option that will create the .gdignore file on the desired directory.
Note: For the files to get exported on these dirs you have to include a filter for them on the "Filters to export non-resources files".

dont_import_scr

### CODE REVIEW

**Comment by mrcdk** [comment]
There's already a file that you can add to a folder so it won't be imported. The file is `.gdignore` http://docs.godotengine.org/en/3.0/getting_started/workflow/project_setup/project_organization.html?highlight=gdignore#importing

**Comment by marcelofg55** [comment]
@mrcdk I tried with that first but if you add that `.gdignore` the directory is not shown anymore on the dock

**Comment by marcelofg55** [comment]
So I though about this and decided it will be better to use the existing .gdignore logic, made the changes on the PR already :).

File: editor/filesystem_dock.cpp [hunk 1]
```diff
@@ -1314,6 +1327,7 @@ void FileSystemDock::_dir_rmb_pressed(const Vector2 &p_pos) {
 		folder_options->add_separator();
 		folder_options->add_item(TTR("New Folder.."), FOLDER_NEW_FOLDER);
 		folder_options->add_item(TTR("Show In File Manager"), FOLDER_SHOW_IN_EXPLORER);
+		folder_options->add_item(TTR("Don't Import This Directory"), FOLDER_DONT_IMPORT);
```
Comment (akien-mga) [inline editor/filesystem_dock.cpp]: "Exclude From Import" might be better.

Comment (akien-mga) [reply editor/filesystem_dock.cpp]: And might require an option to un-blacklist folders..

File: editor/editor_file_system.cpp [hunk 2]
```diff
@@ -579,8 +579,6 @@ void EditorFileSystem::_scan_new_dir(EditorFileSystemDirectory *p_dir, DirAccess
 
 			if (FileAccess::exists(cd.plus_file(f).plus_file("project.godot"))) // skip if another project inside this
 				continue;
-			if (FileAccess::exists(cd.plus_file(f).plus_file(".gdignore"))) // skip if another project inside this
```
Comment (akien-mga) [inline editor/editor_file_system.cpp]: The previous logic meant to exclude all folders with a `.gdignore` file, your change seems to only take into account the current directory?

**Comment by akien-mga** [comment]
I'm not sure this is really practical, you can't keep track of what got excluded or not, and adding an icon in the filesystem to show it/toggle it would be overkill IMO.

I still think that there's a need for an Import Manager in the project settings dialog, that would let you tick/untick in a tree view what you want to import or not, without having to pollute the filesystem with `.gdignore` files.

**Comment by reduz** [comment]
Problem of this is that once excluded, you have to delete the file to bring it back. Maybe the dialog should be like "Create a .gdignore file in this directory? (needs to be removed manually to undo this)"

**Comment by akien-mga** [comment]
I still think it's not the right approach, I agree about the need to expose a system for this, but a one-way exclusion without possibility to review what was exclude and modify it is bad UX.

I'm still quite convinced that we need a real Import Manager that lets you define what folders should be excluded, and the default import settings for each file type found in your filesystem with possibility to force reimport.

**Pull request closed without merging**